core/rend/gles/gles.cpp

```diff
@@ -340,8 +356,13 @@ const char* ModifierVolumeShader =
 uniform lowp float sp_ShaderColor; \n\
 /* Vertex input*/ \n\
 void main() \n\
-{ \n\
-	" FRAGCOL "=vec4(0.0, 0.0, 0.0, sp_ShaderColor); \n\
+{ \n"
+#ifndef GLES
+	"\
+	highp float w = gl_FragCoord.w * 100000.0; \n\
+	gl_FragDepth = log2(1.0 + w) / 34.0; \n"
```

Comment:
I guess those numbers need some tuning from time to time? I used a (somewhat) similar technique for nullDC, though clamped the exponent part to make sure everything fits - https://github.com/skmp/nulldc/blob/master/nulldc/plugins/drkPvr/ps_hlsl.fx#L143

Reply:
Yes, redream was using log2(1 + w * 100) / 24 originally.
I bumped it up as I found one case of loss of precision on very small z.